testtools/iothub_test/src/iothub_account.c

```diff
@@ -0,0 +1,1149 @@
+                    iothub_account_info = NULL;
+                }*/
+                else
+                {
+                    if (retrieveConnStringInfo(iothub_account_info) != 0)
+                    {
+                        LogError("retrieveConnStringInfo failed.\r\n");
+                        free(iothub_account_info);
+                        iothub_account_info = NULL;
+                    }
+                    else
+                    {
+                        iothub_account_info->iothub_service_client_auth_handle = IoTHubServiceClientAuth_CreateFromConnectionString(iothub_account_info->connString);
+                        if (iothub_account_info->iothub_service_client_auth_handle == NULL)
+                        {
+                            LogError("IoTHubServiceClientAuth_CreateFromConnectionString failed\r\n");
+                            free(iothub_account_info);
+                            iothub_account_info = NULL;
+                        }
+                        else
+                        {
+                            iothub_account_info->iothub_messaging_handle = IoTHubMessaging_LL_Create(iothub_account_info->iothub_service_client_auth_handle);
+                            if (iothub_account_info->iothub_messaging_handle == NULL)
+                            {
+                                LogError("IoTHubMessaging_LL_Create failed\r\n");
+                                IoTHubAccount_deinit(iothub_account_info);
+                                iothub_account_info = NULL;
+                            }
+                            else
+                            {
+                                iothub_account_info->iothub_registrymanager_handle = IoTHubRegistryManager_Create(iothub_account_info->iothub_service_client_auth_handle);
+                                if (iothub_account_info->iothub_registrymanager_handle == NULL)
+                                {
+                                    LogError("IoTHubRegistryManager_Create failed\r\n");
+                                    IoTHubAccount_deinit(iothub_account_info);
+                                    iothub_account_info = NULL;
+                                }
+                                else
+                                {
+                                    if (provisionDevices(iothub_account_info, IOTHUB_ACCOUNT_AUTH_CONNSTRING, iothub_account_info->sasDevices, iothub_account_info->number_of_sas_devices) != 0)
+                                    {
+                                        LogError("Failed to create the SAS device(s)\r\n");
+                                        IoTHubAccount_deinit(iothub_account_info);
+                                        iothub_account_info = NULL;
+                                    }
+                                    //else if (provisionDevice(iothub_account_info, IOTHUB_ACCOUNT_AUTH_X509, &iothub_account_info->x509Device) != 0)
```

Comment:
Should this function have authentication_method argument passed down so less changes for future?